compiler/rustc_infer/src/infer/combine.rs

```diff
@@ -125,7 +125,9 @@ impl<'tcx> InferCtxt<'tcx> {
                 bug!()
             }
 
-            (_, ty::Alias(AliasKind::Projection, _)) | (ty::Alias(AliasKind::Projection, _), _)
+            // FIXME(fmease): Is this correct?
+            (_, ty::Alias(AliasKind::Projection | AliasKind::Inherent, _))
```

Comment:
Yes, technically, though I'm not sure if the new solver is adjusted to handle projection goals with IATs, so it'll ICE with `-Ztrait-solver=next`.